### PR TITLE
Initial infrastructure for client side event handling

### PR DESCRIPTION
**Pull request by tombuMS**

### CODE REVIEW

**Comment by eanders-MS** [comment]
Ok looks good so far!

File: MREUnityRuntime/MREUnityRuntimeLib/Behaviors/BehaviorHandlerBase.cs [hunk 1]
```diff
@@ -45,6 +48,17 @@ protected void RegisterActionHandler(MWAction action, string name)
         {
             var handler = new BehaviorActionHandler(((IBehaviorHandler)this).BehaviorType, name, _appRef, _attachedActorId);
             action.Handler = handler;
+            _actionHandlers[name] = handler;
```
Comment (eanders-MS) [inline MREUnityRuntime/MREUnityRuntimeLib/Behaviors/BehaviorHandlerBase.cs]: >name [](start = 28, length = 4)

Do you need to lower-case this (and in GetActionHandler, below)? The value coming from the JSON payload will be lowercase.

Comment (eanders-MS) [reply MREUnityRuntime/MREUnityRuntimeLib/Behaviors/BehaviorHandlerBase.cs]: But the value from nameof(<enum value>) won't be, I believe.

---
In reply to: [316891965](https://github.com/microsoft/mixed-reality-extension-unity/pull/105#discussion_r316891965) [](ancestors = 316891965)

Comment (tombuMS) [reply MREUnityRuntime/MREUnityRuntimeLib/Behaviors/BehaviorHandlerBase.cs]: nameof does return it in the original case.  Good catch.

**Comment by tombuMS** [comment]
Created a regular PR against feature/triggered-actions

**Pull request closed without merging**